src/f4enix/input/MCNPinput.py

```diff
@@ -1339,12 +1330,85 @@ def hash_multiple_cells(self, hash_dict: dict[int, int]) -> None:
                 cell = self.cells[str(cell_num)]
                 self.hash_cell(cell, hash_id, inplace=True)
 
+    @staticmethod
+    def cell_union_function(
+        cell_list: list[parser.Card],
+        new_cell_num: int = None,
+    ) -> parser.Card:
+        """Given a list of cells, it creates a new cell that is the union of
+        the cells in the list.
+
+        Parameters
+        ----------
+        cell_list : list[parser.Card]
+            list of cells to be united
+        new_cell_num : int, optional
+            new number of the union cell, by default None
+        """
+
+        if new_cell_num is None:
+            cel = cell_list[0]
+            new_cell_num = cel.values[0][0]
+
+        new_cell = cell_list[0]
+
+        for cell in cell_list[1:]:
```

Comment:
As discussed please remove this method and just put this for loop in the cells union method.